src/main/java/edu/harvard/iq/dataverse/DvObjectServiceBean.java

```diff
@@ -389,4 +395,18 @@ public String generateNewIdentifierByStoredProcedure() {
         return (String) query.getOutputParameterValue(1);
     }
     
+    /** Backward-compatibility method to get the effective pid generator for a DvObjectContainer.
+     * If the dvObjectContainer method fails, this method will check for the old global default settings.
+     * 
+     * If/when those are no longer supported, this method can be removed.
+     * 
+     */
+    public PidProvider getEffectivePidGenerator(DvObjectContainer dvObjectContainer) {
```

Comment:
This is a backward compatibility method. IMHO it should be flagged `@Deprecated` and have the deprecation Javadoc added, so it gets highlighted for later removal.